### PR TITLE
memory: change memcpy to memcpy_s calls

### PR DESCRIPTION
**Pull request by dabekjakub**
Changed calls to memcpy to memcpy_s - with proper context. After this change using memcpy_s should be default. 
And memcpy should be used sparingly only with a good reason.

Signed-off-by: Jakub Dabek <jakub.dabek@linux.intel.com>

### CODE REVIEW

**Comment by dabekjakub** [comment]
@lgirdwood Could You please help me understand how host building works ? I have problems with figuring out what I am missing. Is it taking defines from baytrail platform/mailbox.h ?

**Comment by xiulipan** [comment]
@dabekjakub 
I think the host build used the include files here
https://github.com/thesofproject/sof/tree/master/src/library/include/platform/memory.h
@ranj063 can you confirm?

**Comment by lgirdwood** [comment]
@dabekjakub host build will use libc versions of these functions. We originally prefixed with an r and wrapped on host e.g. rfree()

File: src/audio/eq_fir.c [hunk 1]
```diff
@@ -175,7 +175,7 @@ static void eq_fir_s16_passthrough(struct fir_state_32x16 fir[],
 	int16_t *dest = (int16_t *)sink->w_ptr;
 	int n = frames * nch;
 
-	memcpy(dest, src, n * sizeof(int16_t));
+	memcpy_s(dest, n * sizeof(int16_t), src, n * sizeof(int16_t));
```
Comment (mmaka1) [inline src/audio/eq_fir.c]: This should rather check the distance between r_ptr/w_ptr and their respective end of buffers.

File: src/audio/eq_fir.c [hunk 2]
```diff
@@ -187,7 +187,7 @@ static void eq_fir_s32_passthrough(struct fir_state_32x16 fir[],
 	int32_t *dest = (int32_t *)sink->w_ptr;
 	int n = frames * nch;
 
-	memcpy(dest, src, n * sizeof(int32_t));
+	memcpy_s(dest, n * sizeof(int32_t), src, n * sizeof(int32_t));
```
Comment (mmaka1) [inline src/audio/eq_fir.c]: This should rather check the distance between r_ptr/w_ptr and their respective end of buffers. Pls check other similar cases in eq_... Some single comp_buffer_copy(struct comp_buffer *source) that performs necessary checks similar to comp_buffer_can_copy_bytes() and calls memcpy_s() could be nice.

**Comment by dabekjakub** [comment]
Closing - will separate it into more prs for clarity.

**Pull request closed without merging**